### PR TITLE
Extract provider install to a COMMAND

### PR DESCRIPTION
**Pull request by Itxaka**
This means that its reusable from any target and can be used without
bringing the current BASE_IMAGE

This command produces a rootfs that can be applied over any base image from any target in any repo. And it contains the basics for the provider to work.

Signed-off-by: Itxaka <itxaka.garcia@spectrocloud.com>

### CODE REVIEW

File: Earthfile [hunk 1]
```diff
@@ -140,16 +140,35 @@ dist:
...
+    ELSE
+        ARG OS_VERSION=${KAIROS_VERSION}
+    END
+
+    ARG OS_ID
+    ARG OS_NAME=${OS_ID}-${FLAVOR}
+    ARG OS_REPO=quay.io/kairos/${VARIANT}-${FLAVOR}
+    ARG OS_LABEL=latest
+
+    DO kairos+OSRELEASE --BUG_REPORT_URL="https://github.com/kairos-io/kairos/issues/new/choose" --HOME_URL="https://github.com/kairos-io/provider-kairos" --OS_ID=${OS_ID} --OS_LABEL=${OS_LABEL} --OS_NAME=${OS_NAME} --OS_REPO=${OS_REPO} --OS_VERSION=${OS_VERSION}-k3s${K3S_VERSION} --GITHUB_REPO="kairos-io/provider-kairos" --VARIANT=${VARIANT}
```
Comment (mauromorales) [inline Earthfile:159]: I guess earthly can handle this back and forth between repos, cool ... (starting form kairos repo) `earthly +some-target` -> `provider-kairos+PROVIDER_INSTALL` -> `kairos+OSRELEASE`

Comment (Itxaka) [reply Earthfile:159]: well, kairos target will only use the PROVIDER_INSTALL command, so it doesnt trigger this, so its only some target -> kairos+OSRELEASE. Im not even sure that would work.

Maybe we should move some of the generic targets to a different repo like earthly-utils or something and have our shared commands there so we avoid recursive deps...

**Comment by Itxaka** [comment]
This needs, [kairos-io/kairos@`e68aa94` (#1592)](https://github.com/kairos-io/kairos/pull/1592/commits/e68aa94034ae148cb5301454ac0b7b64b0264592) to work but that PR needs this PR to work.

So we need to merge first one then the other :D

File: Earthfile (hunk 1 above)
Comment (mauromorales) [reply Earthfile:162]: yeah good idea

**Pull request merged**